src/ai/backend/common/clients/valkey_client/valkey_artifact_registries/client.py

```diff
@@ -106,136 +105,75 @@ async def ping(self) -> None:
-
-    @valkey_artifact_registries_resilience.apply()
-    async def get_huggingface_registry(
-        self,
-        registry_id: uuid.UUID,
-    ) -> Optional[HuggingFaceRegistryStatefulData]:
-        """
-        Get cached HuggingFace registry data.
-
-        :param registry_id: The UUID of the HuggingFace registry.
-        :return: The cached registry data or None if not found.
-        """
-        key = self._make_registry_key(ArtifactRegistryType.HUGGINGFACE, registry_id)
-        value = await self._client.client.get(key)
-        if value is None:
-            log.debug("Cache miss for HuggingFace registry {}", registry_id)
-            return None
-
-        json_value = value.decode()
-        data = load_json(json_value)
-        return HuggingFaceRegistryStatefulData.from_dict(data)
+        return f"artifact_registries.{registry_id}"
 
     @valkey_artifact_registries_resilience.apply()
-    async def delete_huggingface_registry(
+    async def set_registry(
         self,
         registry_id: uuid.UUID,
-    ) -> bool:
-        """
-        Delete cached HuggingFace registry data.
-
-        :param registry_id: The UUID of the HuggingFace registry.
-        :return: True if the key was deleted, False otherwise.
-        """
-        key = self._make_registry_key(ArtifactRegistryType.HUGGINGFACE, registry_id)
-        result = await self._client.client.delete([key])
-        deleted = result > 0
-        if deleted:
-            log.debug("Deleted cached HuggingFace registry data for {}", registry_id)
-        return deleted
-
-    @valkey_artifact_registries_resilience.apply()
-    async def set_reservoir_registry(
-        self,
-        registry_id: uuid.UUID,
-        registry_data: ReservoirRegistryStatefulData,
+        registry_data: T,
     ) -> None:
         """
-        Cache Reservoir registry data.
+        Cache registry data with independent expiry.
 
-        :param registry_id: The UUID of the Reservoir registry.
+        :param registry_id: The UUID of the registry.
         :param registry_data: The registry data to cache.
         """
-        key = self._make_registry_key(ArtifactRegistryType.RESERVOIR, registry_id)
-        value = dump_json_str(dataclasses.asdict(registry_data))
+        key = self._make_registry_key(registry_id)
+        value = dump_json_str(registry_data.to_dict())
         await self._client.client.set(
             key=key,
             value=value,
             expiry=ExpirySet(ExpiryType.SEC, _EXPIRATION),
         )
-        log.debug("Cached Reservoir registry data for {}", registry_id)
+        log.debug("Cached registry data for registry_id={}", registry_id)
 
     @valkey_artifact_registries_resilience.apply()
-    async def get_reservoir_registry(
+    async def get_registry(
         self,
         registry_id: uuid.UUID,
-    ) -> Optional[ReservoirRegistryStatefulData]:
+        data_class: type[T],
+    ) -> Optional[T]:
```

Comment:
It's not really a method that looks very good.
Rather, let Redis itself return the mapping and have the higher level process it.